controllers/operatorpolicy_controller.go

```diff
@@ -790,3 +828,211 @@ func (r *OperatorPolicyReconciler) mergeObjects(
 
 	return updateNeeded, false, nil
 }
+
+func (r *OperatorPolicyReconciler) handleInstallPlan(
+	ctx context.Context, policy *policyv1beta1.OperatorPolicy, sub *operatorv1alpha1.Subscription,
+) error {
+	if sub == nil {
+		err := r.updateStatus(ctx, policy, noInstallPlansCond, noInstallPlansObj(""))
+		if err != nil {
+			return fmt.Errorf("error updating status when the subscription is nil: %w", err)
+		}
+
+		return nil
+	}
+
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	foundInstallPlans, err := r.DynamicWatcher.List(
+		watcher, installPlanGVK, sub.Namespace, labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error listing InstallPlans: %w", err)
+	}
+
+	ownedInstallPlans := make([]unstructured.Unstructured, 0, len(foundInstallPlans))
+
+	for _, installPlan := range foundInstallPlans {
+		for _, owner := range installPlan.GetOwnerReferences() {
+			match := owner.Name == sub.Name &&
+				owner.Kind == subscriptionGVK.Kind &&
+				owner.APIVersion == subscriptionGVK.GroupVersion().String()
+			if match {
+				ownedInstallPlans = append(ownedInstallPlans, installPlan)
+
+				break
+			}
+		}
+	}
+
+	// InstallPlans are generally kept in order to provide a history of actions on the cluster, but
+	// they can be deleted without impacting the installed operator. So, not finding any should not
+	// be considered a reason for NonCompliance.
+	if len(ownedInstallPlans) == 0 {
+		err := r.updateStatus(ctx, policy, noInstallPlansCond, noInstallPlansObj(sub.Namespace))
+		if err != nil {
+			return fmt.Errorf("error updating status when no relevant InstallPlans were found: %w", err)
+		}
+
+		return nil
+	}
+
+	OpLog := ctrl.LoggerFrom(ctx)
+	relatedInstallPlans := make([]policyv1.RelatedObject, len(ownedInstallPlans))
+	ipsRequiringApproval := make([]unstructured.Unstructured, 0)
```

Comment:
Why this is size 0?

Reply:
I just prefer this to an alternate syntax like `ipsRequiringApproval := [] unstructured.Unstructured{}`.

It needs to be length 0 to start so that we can `append` an unknown number of items to it.